lib/wallet/test/unit/Cardano/Wallet/Shelley/TransactionSpec.hs

```diff
@@ -3736,29 +3737,58 @@ estimateSignedTxSizeSpec = describe "estimateSignedTxSize" $ do
         -> ByteString
         -> Cardano.Tx era
         -> IO ()
-    test _name bs tx@(Cardano.Tx (body :: Cardano.TxBody era) _) = do
+    test _name bs cardanoTx@(Cardano.Tx body _) = do
         let pparams = Write.pparamsLedger $ mockPParamsForBalancing @era
-            utxo = utxoPromisingInputsHaveVkPaymentCreds body
-            witCount = estimateKeyWitnessCount (Write.fromCardanoUTxO utxo) body
+            witCount dummyAddr = estimateKeyWitnessCount
+                    (Write.fromCardanoUTxO
+                        $ utxoPromisingInputsHaveAddress dummyAddr body)
+                    body
+            era = recentEra @era
 
-            ledgerTx :: Write.Tx (Write.ShelleyLedgerEra era)
-            ledgerTx = Write.fromCardanoTx @era tx
+            tx :: Write.Tx (Write.ShelleyLedgerEra era)
+            tx = Write.fromCardanoTx @era cardanoTx
 
             noScripts = Write.withConstraints (recentEra @era) $
-                Map.null $ ledgerTx ^. witsTxL . scriptTxWitsL
+                Map.null $ tx ^. witsTxL . scriptTxWitsL
             noBootWits = Write.withConstraints (recentEra @era) $
-                Set.null $ ledgerTx ^. witsTxL . bootAddrTxWitsL
+                Set.null $ tx ^. witsTxL . bootAddrTxWitsL
             testDoesNotYetSupport x =
                 pendingWith $ "Test setup does not work for txs with " <> x
 
+            signedBinarySize = TxSize $ fromIntegral $ BS.length bs
+
         case (noScripts, noBootWits) of
                 (True, True) -> do
-                    estimateSignedTxSize pparams witCount body
-                        `shouldBe`
-                        TxSize (fromIntegral (BS.length bs))
+                    estimateSignedTxSize era pparams (witCount vkCredAddr) tx
+                        `shouldBeInclusivelyWithin`
+                        (signedBinarySize - correction, signedBinarySize)
                 (False, False) -> testDoesNotYetSupport "bootstrap wits + scripts"
-                (True, False) -> testDoesNotYetSupport "bootstrap wits"
+                (True, False) ->
+                    estimateSignedTxSize era pparams (witCount bootAddr) tx
+                        `shouldBeInclusivelyWithin`
+                        ( signedBinarySize - correction
+                        , signedBinarySize + TxSize 45
```

Comment:
Where does this value of `45` come from?

Can we make it a named constant, with its own documentation to describe its provenance?

Reply:
6000e90fc8aa10d288652acd23c43a4b2c9e2f7f